source/common/mongo/proxy.cc

```diff
@@ -36,38 +42,46 @@ void AccessLog::logMessage(const Message& message, bool full,
 }
 
 ProxyFilter::ProxyFilter(const std::string& stat_prefix, Stats::Scope& scope,
-                         Runtime::Loader& runtime, AccessLogSharedPtr access_log)
+                         Runtime::Loader& runtime, AccessLogSharedPtr access_log,
+                         FaultConfigSharedPtr fault_config)
     : stat_prefix_(stat_prefix), scope_(scope), stats_(generateStats(stat_prefix, scope)),
-      runtime_(runtime), access_log_(access_log) {
-
-  if (!runtime_.snapshot().featureEnabled("mongo.connection_logging_enabled", 100)) {
+      runtime_(runtime), access_log_(access_log), fault_config_(fault_config) {
+  if (!runtime_.snapshot().featureEnabled(CONNECTION_LOGGING_ENABLED_KEY, 100)) {
     // If we are not logging at the connection level, just release the shared pointer so that we
     // don't ever log.
     access_log_.reset();
   }
 }
 
-ProxyFilter::~ProxyFilter() {}
+ProxyFilter::~ProxyFilter() { ASSERT(!delay_timer_); }
 
 void ProxyFilter::decodeGetMore(GetMoreMessagePtr&& message) {
```

Comment:
what about decodeReply?

Reply:
Is there a better place to add the cal to tryInjectDelay so we don't have too start every method with `tryInjectDelay()`

Reply:
no need for Reply as it's the response for the original injected request (query, insert, etc)

Reply:
let's see for the tryInjectDelay() place